be/src/util/compression/block_compression.cpp

```diff
@@ -718,13 +718,27 @@ class ZlibBlockCompression : public BlockCompressionCodec {
 
 class ZstdBlockCompression final : public BlockCompressionCodec {
 public:
-    ZstdBlockCompression() : BlockCompressionCodec(CompressionTypePB::ZSTD) {}
+    ZstdBlockCompression() : BlockCompressionCodec(CompressionTypePB::ZSTD), _level(-1) {}
+    ZstdBlockCompression(int level) : BlockCompressionCodec(CompressionTypePB::ZSTD), _level(level) {}
 
     static const ZstdBlockCompression* instance() {
         static ZstdBlockCompression s_instance;
         return &s_instance;
     }
 
+    static const ZstdBlockCompression* instance(int level) {
+        DCHECK(level >= 1 && level <= 22);
```

Comment:
remove this DCHECK and return nullptr if level is out of range.

Never take DCHECK as a the parameter error handling.

Reply:
Fixed